File: cypress-framework/cypress/pages/RegisterPage.js
class RegisterPage {
  static visit()         { cy.visit("/index.php?route=account/register&language=en-gb"); return this; }
  static fill({firstname, lastname, email, telephone, password}) {
    cy.get("#input-firstname").type(firstname);
    cy.get("#input-lastname").type(lastname);
    cy.get("#input-email").type(email);
    cy.get("#input-telephone").type(telephone);
    cy.get("#input-password").type(password, { log:false });
    return this;
  }
  static agree()  { cy.get("input[name=agree]").check({ force: true }); return this; }
  static submit() { cy.get("button[type=submit]").click(); return this; }
}
export default RegisterPage;

File: selenium-framework/pages/register_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage


class RegisterPage(BasePage):
    FIRSTNAME  = (By.ID, "input-firstname")
    LASTNAME   = (By.ID, "input-lastname")
    EMAIL      = (By.ID, "input-email")
    TELEPHONE  = (By.ID, "input-telephone")
    PASSWORD   = (By.ID, "input-password")
    AGREE      = (By.CSS_SELECTOR, "input[name='agree']")
    SUBMIT     = (By.CSS_SELECTOR, "button[type='submit']")
    SUCCESS_H  = (By.CSS_SELECTOR, "h1")  # "Your Account Has Been Created!"
    ALERT      = (By.CSS_SELECTOR, ".alert-danger, .text-danger")

    def go(self):
        return self.open("account/register&language=en-gb")

    def register(self, firstname, lastname, email, telephone, password, agree=True):
        self.type(self.FIRSTNAME, firstname)
        self.type(self.LASTNAME,  lastname)
        self.type(self.EMAIL,     email)
        self.type(self.TELEPHONE, telephone)
        self.type(self.PASSWORD,  password)
        if agree:
            self.click(self.AGREE)
        self.click(self.SUBMIT)
        return self

    def success_visible(self) -> bool:
        return self.is_visible(self.SUCCESS_H) and "Created" in self.text_of(self.SUCCESS_H)
